packages/ember-glimmer/lib/environment.js

```diff
@@ -290,6 +292,8 @@ export default class Environment extends GlimmerEnvironment {
       return (vm, args) => SimpleHelperReference.create(helper.compute, args);
     } else if (helper.isHelperFactory) {
       return (vm, args) => ClassBasedHelperReference.create(helper, vm, args);
+    } else if (helper.glimmerNativeHelper) {
+      return helper.glimmerNativeHelper;
```

Comment:
@ef4 I think we can simplify this by making internal helpers just a function with this interface: https://github.com/tildeio/glimmer/blob/master/packages/glimmer-runtime/lib/environment.ts#L239-L241

Then you wouldn't need the `isInternalHelper` or `glimmerNativeHelper` flags – which are problematic because a userland helper can pretend to be one of those just by assigning that property on them.

Instead:

```
let builtInHelper = this.builtInHelpers[name];

if (builtInHelper) {
  return builtInHelper;
}

// do userland lookups...
```